lib/ohai/plugins/network.rb

```diff
@@ -159,7 +166,9 @@ def network_contains_address(address_to_match, ipaddress, iface)
           Ohai::Log.debug("unable to detect ip6address")
         else
           ip6address r["ip"]
-          if r["mac"] and macaddress.nil? and ipaddress.nil?
+          # don't overwrite macaddress set by "#{os}::network" plugin
+          # and also
+          if r["mac"] and macaddress.nil? and ( ipaddress.nil? || ipaddress == "127.0.0.1" )
```

Comment:
I'm in agreement that this isn't the long term desired behavior. It's purely for backwards compatibility so we don't overwrite the v4 mac with a v6 on every run.  At the moment we do have the expectation through the specs that v4 takes precedence, but not setting the v6 mac because ipaddress was set from lo seems a bit silly.  

A flag would be useful here, but I wonder if we're running into a fundamental limitation of the single macaddress attribute that contains both the mac of the v6 interface and the v4 interface.